bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1982,26 +1982,43 @@ private static boolean checkIsLikeTupleType(Object sourceValue, BTupleType targe
         }
 
         ArrayValue source = (ArrayValue) sourceValue;
-        if (source.size() != targetType.getTupleTypes().size()) {
+        List<BType> targetTypes = new ArrayList<>(targetType.getTupleTypes());
+        int sourceTypeSize = source.size();
+        int targetTypeSize = targetTypes.size();
+        BType targetRestType = targetType.getRestType();
+
+        if (targetRestType == null && sourceTypeSize != targetTypeSize) {
+            return false;
+        }
+        if (targetRestType != null && sourceTypeSize <= targetTypeSize) {
             return false;
         }
 
-        int bound = source.size();
-        for (int i = 0; i < bound; i++) {
-            BType elementType = getArrayElementType(source, i);
-            if (BTypes.isValueType(elementType)) {
-                if (!checkIsType(elementType, targetType.getTupleTypes().get(i), new ArrayList<>())) {
-                    return false;
-                }
-            } else {
-                if (!checkIsLikeType(source.getRefValue(i), targetType.getTupleTypes().get(i), unresolvedValues,
-                        allowNumericConversion)) {
-                    return false;
-                }
+        for (int i = 0; i < targetTypeSize; i++) {
+            if (!checkTupleElementType(getArrayElementType(source, i), targetTypes.get(i), source.getRefValue(i),
```

Comment:
Why do we have to use special methods `checkTupleElementType` and `getArrayElementType` here and in the next for loop? 

Can't (shouldn't) we directly call 
`checkIsLikeType(source.getRefValue(i), targetTypes.get(i), unresolvedValues, allowNumericConversion)`?

Reply:
Changed.